addons/website_sale/views/templates.xml

```diff
@@ -1849,7 +1849,7 @@
                         <b>Communication: </b><span t-esc='order.reference'/>
                     </div>
                     <div t-if="payment_tx_id.acquirer_id.sudo().qr_code">
-                        <t t-set="qr_code" t-value="payment_tx_id.acquirer_id.sudo().journal_id.bank_account_id.build_qr_code_base64(order.amount_total,payment_tx_id.reference, None, payment_tx_id.currency_id, payment_tx_id.partner_id)"/>
+                        <t t-set="qr_code" t-value="payment_tx_id.company_id.partner_id.bank_ids[:1].build_qr_code_base64(order.amount_total,payment_tx_id.reference, None, payment_tx_id.currency_id, payment_tx_id.partner_id)"/>
```

Comment:
What about `payment_tx_id.payment_id.journal_id.bank_account_id`?
The `[:1]` seems very arbitrary and wrong in the vast majority of cases.

Not sure what the flow if `payment` is though so I might be wrong.
At the very least, try to get the journal from the `acquirer_id` if you need to get the bank from the company **as a fallback**

Reply:
I talked about this issue with Benj. 
In the case of wire transfer, the bank journal is not involved (We can't even set it on the acquirer anymore). 
So it doesn't make much sense to look in the journal for the account.

For your first point, we have no payment set here yet. (nor invoices)
ABZ proposed getting the journal from the company (bank_journal_ids) to solve this issue. But then again, we just take the first account we find and I think you don't have as much control over the sequence of journals as over the order of sequence of the accounts set on the partners.
What's wrong with [:1]? By curiosity

In the end, Benj was of the idea that all companies should have at least one bank account set, and that we should take the first one we find on the company's partner. If no account has been provided, then it's a configuration issue.
This issue would happen both ways anyway, since setting the account on the bank journal will set it on the company's partner too. So if the company one is empty, the other is most likely empty too. (Unless manually removed/..)

Reply:
> What's wrong with [:1]? By curiosity

That you can have many bank accounts for your company, and it doesn't really make sense to pick the first one, unless it doesn't matter at all, which I believe is not the case for QR-code generation (different countries, currencies, rules might make you want your client to pay on one or another account)

Reply:
Also, the current fix is breaking all other payment methods